src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/SafeEvpPKeyHandle.OpenSsl.cs

```diff
@@ -4,13 +4,20 @@
 using System.Diagnostics;
 using System.Runtime.InteropServices;
 using System.Runtime.Versioning;
+using Microsoft.Win32.SafeHandles;
 
 namespace System.Security.Cryptography
 {
     public sealed partial class SafeEvpPKeyHandle : SafeHandle
     {
         internal static readonly SafeEvpPKeyHandle InvalidHandle = new SafeEvpPKeyHandle();
 
+        /// <summary>
+        /// In some cases like when a key is loaded from a provided, the key may have an associated data
+        /// we need to keep alive for the lifetime of the key. This field is used to track that data.
+        /// </summary>
+        internal IntPtr ExtraHandle { get; private set; } = IntPtr.Zero;
```

Comment:
Are there cases where it's not a provider, or should it be called ProviderHandle?

Reply:
It's really provider handle + library context - I don't feel particularly strongly about the name but ProviderHandle seems a bit off even if it's the only API which makes this non-null